lib/Doctrine/DBAL/Connection.php

```diff
@@ -615,9 +615,14 @@ private function addIdentifierCondition(
         $platform = $this->getDatabasePlatform();
 
         foreach ($identifier as $columnName => $value) {
-            if ($value === null) {
-                $conditions[] = $platform->getIsNullExpression($columnName);
-                continue;
+            if (!is_array($value)) {
+                if ($value === null) {
+                    $conditions[] = $platform->getIsNullExpression($columnName);
+                    continue;
+                } elseif (strtoupper($value) === 'NOT NULL') {
```

Comment:
How is it supposed to work with a string value "NOT NULL"?

Reply:
$this->connection->update('connections', ['timezone' => 'my zone'], ['timezone' => 'NOT NULL']);

Reply:
What if I want to check that the value is not equal to the string "NOT NULL"?

Reply:
Ok it's bad idea. I think we should rather look at columnName